includes/Modules/Analytics_4/Settings.php

```diff
@@ -95,23 +122,76 @@ protected function get_sanitize_callback() {
 				if ( isset( $option['useSnippet'] ) ) {
 					$option['useSnippet'] = (bool) $option['useSnippet'];
 				}
+
 				if ( isset( $option['googleTagID'] ) ) {
 					if ( ! preg_match( '/^(G|GT|AW)-[a-zA-Z0-9]+$/', $option['googleTagID'] ) ) {
 						$option['googleTagID'] = '';
 					}
 				}
-				if ( isset( $option['googleTagAccountID'] ) ) {
-					if ( ! is_numeric( $option['googleTagAccountID'] ) || ! $option['googleTagAccountID'] > 0 ) {
-						$option['googleTagAccountID'] = '';
-					}
-				}
-				if ( isset( $option['googleTagContainerID'] ) ) {
-					if ( ! is_numeric( $option['googleTagContainerID'] ) || ! $option['googleTagContainerID'] > 0 ) {
-						$option['googleTagContainerID'] = '';
+
+				$numeric_properties = array( 'googleTagAccountID', 'googleTagContainerID' );
+				foreach ( $numeric_properties as $numeric_property ) {
+					if ( isset( $option[ $numeric_property ] ) ) {
+						if ( ! is_numeric( $option[ $numeric_property ] ) || ! $option[ $numeric_property ] > 0 ) {
+							$option[ $numeric_property ] = '';
+						}
 					}
 				}
 			}
+
 			return $option;
 		};
 	}
+
+	/**
+	 * Updates settings to set the current user as an owner of the module if settings contain owned keys.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $settings The module settings.
+	 */
+	protected function maybe_set_owner_id( $settings ) {
+		$keys = $this->get_owned_keys();
+		if ( count( array_intersect( array_keys( $settings ), $keys ) ) > 0 && current_user_can( Permissions::MANAGE_OPTIONS ) ) {
+			$user_id = get_current_user_id();
+			$this->merge( array( 'ownerID' => $user_id ) );
```

Comment:
Maybe we shouldn't set `ownerID` at all on the `Analytics_4` settings as for one thing, it will be out of sync with `Analytics` until `Analytics_4` owned settings are updated, and for another, it might just be a bit less clear when looking at the settings, having to remember that the Analytics one is the source of truth.